### ISSUE TITLE
question: how can I use tensorly for a vectorized input?

### ISSUE THREAD

**Issue by salimoha** [tensorly/tensorly#86]
My input is a 30-dimensional variable. How can use tensorly to perform dimension reduction and reconstruct my input again? e.g., similar to PCA. thanks

**Comment by JeanKossaifi** [comment]
I am not sure what you want to do -- decompose a vector?
If it is a vectorized tensor, why not keep the original tensor and decompose this?

If you want the analogous of PCA for higher orders, your best bet is the [Tucker decomposition](http://tensorly.org/stable/modules/generated/tensorly.decomposition.tucker.html#tensorly.decomposition.tucker).

**Comment by salimoha** [comment]
I followed the instructions; however, if you run the code below, you see a large Error for reconstruction of the original vector. How is it possible to reduce the error? 

```
import tensorly as tl
from scipy.misc import face, imresize
from tensorly.decomposition import parafac
from tensorly.decomposition import tucker
from math import ceil

import pandas as pd
import numpy as np

# generate a random data 
df = pd.DataFrame(np.random.randn(15000, 6), columns=list('ABCDEF'))

# Rank of the Tucker decomposition
tucker_rank = [int(15000/2), int(6/2)]
# Tucker decomposition
data = tl.tensor(df, dtype='float64')
core,tucker_factors  = tucker(data, ranks=tucker_rank)
# reconstruct the vecotor 
tucker_approximation = tl.tucker_to_tensor(core, tucker_factors)
Errors = np.sqrt(np.sum((tucker_approximation-data)**2, axis=1))
```

**Comment by JeanKossaifi** [comment]
There are several issues here:
* Your data is 2-D so tensor methods will not bring any improvement (for second order tensors, e.g. matrices, Tucker is essentially an SVD), it seems what you want here is [PCA](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html) 
* You are also decomposing along the samples axis: I am not sure this is what you need or want
* You might want to normalise your error by the norm of the samples
*  You are trying to decompose a matrix with random elements, which will therefore have full rank with probability 1. Since it is a matrix you are decomposing, we also know this rank is the minimum of the number of rows and number of columns (6 in this case).  However, you are decomposing it into a matrix of size (5000, 3) which will have a maximum rank of 3.. You cannot hope to reconstruct the original matrix exactly. You should be able to recover your original matrix exactly by using a rank of 6 for both dimensions in this case.
